src/main/edit_delete/preProcess.ts

```diff
@@ -1,11 +1,47 @@
-const preProcess = (wordList: Array<any>, duration: number): Array<any> => {
-  for (let i = 0; i < wordList.length - 1; i += 1) {
-    wordList[i].duration = wordList[i + 1].start_time - wordList[i].start_time;
-    wordList[i].deleted = false;
+import { Transcription } from 'sharedTypes';
+import JSONTranscription from './JSONTranscription';
+import TranscriptWord from './TranscriptWord';
+
+const preProcessTranscript = (
+  jsonTranscript: any,
```

Comment:
should `jsonTranscript` be of type `Transcript`?

Reply:
Transcript and Word are interfaces to represent the transcription, we made JSONTranscription and TranscriptWord as concrete implementations of these interfaces.

Reply:
Don't think so, as the input is raw JSON. This function makes it into a Transcription type

Reply:
sorry my bad! You're right - I forgot you are taking the raw json from the ml team - not a Transcript object. This is fine 🙂